Cargo.toml

```diff
@@ -0,0 +1,26 @@
+[package]
+name = "gateway-config"
+version = "0.1.0"
+edition = "2024"
+license = "Apache-2.0"
+
+[features]
+default = []
+regenerate = ["dep:tonic-build", "dep:protoc-bin-vendored"]
+
+[dependencies]
+tonic = "0.13"
+prost = "0.13.5"
+tokio = { version = "1.34", features = ["full"] }
+futures = "0.3"
+async-trait = "0.1"
+serde = { version = "1.0", features = ["derive"] }
+serde_json = "1.0"
+
+[dev-dependencies]
+tower = "0.5.2"
+tokio-stream = { version = "0.1", features = ["net"] }
+
+[build-dependencies]
+tonic-build = {version = "0.13", optional = true }
+protoc-bin-vendored = { version = "3.1.0", optional = true }
```

Comment:
If I understand correctly, that's just embedding the `protoc` binary, right? We are already handling it in the Justfile for the Golang code, IMO it'll be better to use it so we're relying on the same version, but it's okay for now. E.g. it's not directly clear which version protoc-bin-vendored is using.

https://github.com/githedgehog/gateway-proto/blob/master/hack/tools.just#L33